### PR TITLE
Validate SSH keys list before view/delete

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
Validate SSH key list in the view / delete actions of the SSH Plugin

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14218

### CODE REVIEW

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 1]
```diff
@@ -130,8 +130,22 @@ const createKeyPair = async (sshkeyManager: SshKeyManager) => {
     }
 };
 
+const getKeys = async (sshKeyManager: SshKeyManager): Promise<cheApi.ssh.SshPair[]> => {
+    const keys: cheApi.ssh.SshPair[] = await sshKeyManager.getAll('vcs');
+    if (!keys || keys.length < 1) {
+        throw new Error('No SSH key pairs has been defined.');
```
Comment (benoitf) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: `SSH Key Pair`
or 
`SSH Key Pairs have`

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 2]
```diff
@@ -130,8 +130,22 @@ const createKeyPair = async (sshkeyManager: SshKeyManager) => {
...
+    return keys;
+};
+
 const deleteKeyPair = async (sshkeyManager: SshKeyManager) => {
-    const keys: cheApi.ssh.SshPair[] = await sshkeyManager.getAll('vcs');
+    let keys: cheApi.ssh.SshPair[];
+    try {
+        keys = await getKeys(sshkeyManager);
+    } catch (error) {
+        theia.window.showWarningMessage(error.message);
```
Comment (benoitf) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: should say that we're trying to delete in warning message

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 3]
```diff
@@ -150,7 +164,13 @@ const deleteKeyPair = async (sshkeyManager: SshKeyManager) => {
...
 
 const viewPublicKey = async (sshkeyManager: SshKeyManager) => {
-    const keys: cheApi.ssh.SshPair[] = await sshkeyManager.getAll('vcs');
+    let keys: cheApi.ssh.SshPair[];
+    try {
+        keys = await getKeys(sshkeyManager);
+    } catch (error) {
+        theia.window.showWarningMessage(error.message);
+        return;
+    }
```
Comment (benoitf) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: we could move to a separate method as we're using twice

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 1 above)
Comment (vinokurig) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: fixed

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 2 above)
Comment (vinokurig) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: done

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 3 above)
Comment (vinokurig) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: done

**Pull request merged**